### ISSUE TITLE
assertion failure in var_summary_get_ref_map

### ISSUE THREAD

**Issue by chergert** [flatpak/flatpak#4245]
After running a `flatpak --user repair`, I tried installing an SDK with Builder (which uses the deprecated installation APIs, but supported nonetheless).

The SDK was org.freedesktop.Sdk//19.04 from flathub.

```
flatpak:ERROR:common/flatpak-variant-impl-private.h:1185:var_summary_get_ref_map: assertion failed: (end <= v.size)
Bail out! flatpak:ERROR:common/flatpak-variant-impl-private.h:1185:var_summary_get_ref_map: assertion failed: (end <= v.size)
```

**Comment by chergert** [comment]
It also fails from the command line

```
[📦christian@fedora build]$ flatpak install --user org.freedesktop.Sdk//19.04 
Looking for matches…
**
flatpak:ERROR:common/flatpak-variant-impl-private.h:1185:var_summary_get_ref_map: assertion failed: (end <= v.size)
Bail out! flatpak:ERROR:common/flatpak-variant-impl-private.h:1185:var_summary_get_ref_map: assertion failed: (end <= v.size)
Aborted (core dumped)
```

**Comment by chergert** [comment]
Update also results in the same crash.

**Comment by smcv** [comment]
Which version of Flatpak is this? This looks similar to a bug that I thought was fixed a while ago.

**Comment by chergert** [comment]
Flatpak 1.10.2 in Fedora 34

**Comment by smcv** [comment]
See also #3503

**Comment by artscoop** [comment]
Same issue since today with Flatpak 1.10.2-1 in Manjaro Linux. Happened around two months ago, but I could fix it. But now, it just fails with `flatpak history`, `flatpak update` and `flatpak install`. `flatpak list` works

**Comment by Gert-dev** [comment]
Never had any issues with flatpak and been using it in a stable way for months, but today I suddenly started getting this issue as well. I just uninstalled a package, and wanted to install it again five minutes later, which I can't seem to do anymore now. Unlike @artscoop some other commands also fail, albeit with other errors.

`flatpak update` and `flatpak install`:

```
Looking for updates…
**
flatpak:ERROR:common/flatpak-variant-impl-private.h:1197:var_summary_get_metadata: assertion failed: (start <= end)
Bail out! flatpak:ERROR:common/flatpak-variant-impl-private.h:1197:var_summary_get_metadata: assertion failed: (start <= end)
fish: Job 1, 'flatpak update' terminated by signal SIGABRT (Abort)
```

`flatpak history`:

```
error: appstream2/x86_64 is not application or runtime
```

`flatpak list` works.

**Comment by mwleeds** [comment]
> Never had any issues with flatpak and been using it in a stable way for months, but today I suddenly started getting this issue as well. I just uninstalled a package, and wanted to install it again five minutes later, which I can't seem to do anymore now. Unlike artscoop some other commands also fail, albeit with other errors.
> 
> `flatpak update` and `flatpak install`:
> 
> ```
> Looking for updates…
> **
> flatpak:ERROR:common/flatpak-variant-impl-private.h:1197:var_summary_get_metadata: assertion failed: (start <= end)
> Bail out! flatpak:ERROR:common/flatpak-variant-impl-private.h:1197:var_summary_get_metadata: assertion failed: (start <= end)
> fish: Job 1, 'flatpak update' terminated by signal SIGABRT (Abort)
> ```
> 

This is fixed in master by #4568

You can work around it by doing `rm ~/.cache/flatpak/system-cache/summaries/*`

> `flatpak history`:
> 
> ```
> error: appstream2/x86_64 is not application or runtime
> ```
> 

The fix for this is available in #4526

**Comment by mwleeds** [comment]
Closing as duplicate of #4127
